### ISSUE TITLE
Create custom settings

### ISSUE THREAD

**Issue by KeizerDev** [KeizerDev/Rhythmbox-Chromecast#3]

**Comment by fossfreedom** [comment]
not sure what this is ... but if its referring to obtaining from the user graphically their IP address and port number then this is being done under #2

**Comment by KeizerDev** [comment]
Yes, actually this should be ip/port/default chromecast(I'm not sure if default chromecast is a good option but yeah) and #2 should be a list of the available chromecasts in a network (pychromecast.get_chromecasts_as_dict().keys())

**Comment by fossfreedom** [comment]
ok - chromecast preferences dialog created - see the preferences branch

you'll need to install the schema

    sudo cp schema/org* /usr/share/glib-2.0/schemas 
    sudo glib-compile-schemas /usr/share/glib-2.0/schemas

Since your main module is "Chromecast.py" - with a capital "C" - the ~/.local/share/rhythmbox/plugins folder must be called "Chromecast" not "chromecast" otherwise the preferences dialog will not work.

You'll see that I've used threading to prevent the GUI freezing when connecting to chromecast.

The "Prefs.py" is no longer used - I've just renamed it for the moment.

**Comment by KeizerDev** [comment]
Great work! I do like to call it chromecast rather than Chromecast. Can you refactor that for me or is it as easy as just renaming the main file? 

The GUI is starting normally but I still think that you should give an indactor of the extension and just list the Chromecast on the main screen and toggle the connection there. In perfect situation we should make a list button next to the player and do it like all Android apps do to toggle the connection.
![](http://b-i.forbesimg.com/scottallison/files/2013/07/Screen-Shot-2013-07-28-at-1.45.30-PM.png)

After a while rhythmbox still freezes but yeah, we are getting closer.

**Comment by fossfreedom** [comment]
quite correct - just need to rename the main module from Chromecast.py to chromecast.py (and change the .plugin module name to match)

You can't add the chromecast icon in the toolbar - but you could add it I suppose to the sidebar.

For the moment I would suggest a button just be added to the sidebar which should allow you to connect or disconnect the chromecast set preferences rather than trying to connect immediately on startup.

Once the crashing is fixed then should revisit this bit to add a "chromecast" icon as you suggest above.
